packages/main/scripts/download-remote-extensions.ts

```diff
@@ -0,0 +1,102 @@
+/*********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************/
+// eslint-disable-next-line import/no-extraneous-dependencies
+import 'reflect-metadata';
+
+import { existsSync } from 'node:fs';
+import { mkdir, rename } from 'node:fs/promises';
+import { tmpdir } from 'node:os';
+import { isAbsolute, join } from 'node:path';
+
+import minimist from 'minimist';
+
+import type { ApiSenderType } from '/@/plugin/api.js';
+import type { Certificates } from '/@/plugin/certificates.js';
+import { ImageRegistry } from '/@/plugin/image-registry.js';
+import type { Proxy } from '/@/plugin/proxy.js';
+import type { Telemetry } from '/@/plugin/telemetry/telemetry.js';
+
+import product from '../../../product.json' with { type: 'json' };
+
+/**
+ * We create _dummy_ classes for the constructor of ImageRegistry
+ * We do not use those classes when calling ImageRegistry#downloadAndExtractImage as the internal logic
+ * uses different node_modules; those classes are needed for other things the class is doing
+ */
+const dummyApiSenderType = {} as unknown as ApiSenderType;
+const dummyTelemetryMock = {} as unknown as Telemetry;
+const dummyCertificateMock = {
+  getAllCertificates: (): undefined => undefined,
```

Comment:
remark: if someones renames `getAllCertificates` to `getAllCertificateList` for instance, the script will fail but we won't know it in the PR checks.

Reply:
Yes, there is clearly room for improvement, we need to refactor the ImageRegistry so it can be consumed as a standalone class to have this fully typed without unknown casting

Reply:
> We do not use those classes when calling ImageRegistry#downloadAndExtractImage as the internal logic
 
question: AFAIK we're using Certificates when calling downloadAndExtractImage

downloadAndExtractImage call getManifest that call getManifestFromURL that call getOptions that call certificates getAllCertificates

so IMHO we can create a real instance of Certificates class